composition-js/src/compose.ts

```diff
@@ -25,20 +25,30 @@ export interface CompositionFailure {
   schema?: undefined;
   supergraphSdl?: undefined;
   hints?: undefined;
+  subgraphSdl?: { [subgraph: string]: string}
 }
 
 export interface CompositionSuccess {
   schema: Schema;
   supergraphSdl: string;
   hints: CompositionHint[];
   errors?: undefined;
+  subgraphSdl?: { [subgraph: string]: string}
 }
 
 export interface CompositionOptions {
   sdlPrintOptions?: PrintOptions;
   allowedFieldTypeMergingSubtypingRules?: SubtypingRule[];
   /// Flag to toggle if satisfiability should be performed during composition
   runSatisfiability?: boolean;
```

Comment:
this would be a breaking change.... but ideally this should move under new `pipelineOptions`?

Reply:
It would be a breaking change to add an extra optional field to the return type?

Reply:
if was thinking about moving this over but yeah if we also add extra option there then yeah that should be fine (and IMHO more consistent, maybe we could deprecate the old field as well)

Reply:
Sorry, which old field do you mean?

Reply:
I thought this PR would make it possible to support the future `HybridComposition`.  Can `federation-rs` call composition with the `piplineOptions` to achieve the goals, no? 🤔

Reply:
Kind of -> using pipeline options would enable us to skip first or last stages but will not allow pick and choose style of integrations.

i.e. given following stages

```js
  pipelineOptions?: {
    runUpgradeSubgraphs?: boolean;
    runValidateSubgraphs?: boolean;
    runComposition?: boolean;
    runSatisfiability?: boolean;
    outputUpgradedSubgraphs?: boolean;
  }
```

How would you integrate it with hybrid composition if you only want to replace subgraph validations with Rust version? Or if you just want to compare single phases?

Reply:
I could move it, but I was thinking that `runSatisfiability` will be more long lived, but the other ones are really only for our use and will be temporary.